core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2788,15 +2788,28 @@ public static double log(BigDecimal d0, BigDecimal d1) {
     return Math.log(d0.doubleValue()) / Math.log(d1.doubleValue());
   }
 
-  /** SQL {@code LOG2(number)} function applied to double values. */
-  public static @Nullable Double log2(double number) {
-    return (number <= 0) ? null : log(number, 2);
+  /** SQL {@code LOG(number, number2)} function applied to double values.
+   * but return null when number2 is 0. */
+  public static @Nullable Double logMysqlSpark(double number, double number2) {
+    return (number <= 0) ? null : log(number, number2);
   }
 
-  /** SQL {@code LOG2(number)} function applied to
-   * BigDecimal values. */
-  public static @Nullable Double log2(BigDecimal number) {
-    return log2(number.doubleValue());
+  /** SQL {@code LOG(number, number2)} function applied to double and BigDecimal values.
```

Comment:
Does this function only exist because MySQL and Spark handle null behavior differently? If so, you shouldn't need to add methods here. Just changed the NullPolicy for that operator.

Reply:
good idea， let me try. I'm very sorry that I didn't describe it clearly in the jira case, I will improve it.
closing.